### DISCUSSION TITLE
Update to 6.7.0 caused error failed to load dynlib/dll

### DISCUSSION THREAD

**Discussion by karam-jaber-intel** [pyinstaller/pyinstaller#8573] [Help]
I updated pyinstaller from 5.13.2 to 6.7.0 and my app is failing on this 

'/tmp/_MEIOaQbCB/app/services/lib/linux/Mylib.so'. Most likely this dynlib/dll was not found when the application was frozen.'

I checked the file in temp and it do exist...what could cause this to happen ?

**Comment by rokm** [comment]
What is the full error traceback?  (i.e., what was the exception from which this one was raised)

**Reply by karam-jaber-intel** [reply]
this is the error together with "/lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found " error.

My application is built on Ubuntu 22 with glibc 2.33 and im running it on Ubuntu18 with glibc 2.27.

Until the Pyinstaller update this worked. after the update i get the GLIBC_2.33 missing error.

Anything has changed with freezing *.so files using pyinstaller onefile approach?

**Reply by rokm** [reply]
> My application is built on Ubuntu 22 with glibc 2.33 and im running it on Ubuntu18 with glibc 2.77.

That won't work. You cannot use binaries compiled against newer version of glibc on systems with older version of glibc (Ubuntu 18 has glibc 2.17).

> Until the Pyinstaller update this worked. after the update i get the GLIBC_2.33 missing error.
> Anything has changed with freezing *.so files using pyinstaller onefile approach?

Is this `GLIBC_2.33' not found` error given directly for `Mylib.so` or for one of its dependencies? 

The only scenario I can imagine where this might have worked before with 5.13.2 is if `Mylib.so` is compiled against old glibc (i.e., Ubuntu 18 one), but has additional dependency that PyInstaller 5.13.2 failed to collect (and thus system copy was used at run-time) whereas PyInstaller 6.x collects it from the build system (and thus this dependency becomes a problem at run-time, being linked against glibc 2.33).

**Reply by rokm** [reply]
I suppose you will have to check both 5.13.2 build and 6.7.0 build, and see if there is any difference in collected .so libraries, especially the ones that `Mylib.so` depends on.

**Reply by karam-jaber-intel** [reply]
Do you know a way to verify if your theory is correct(5.13.2 failed to collect dependency)?

the MyLib.so is same for both builds.

**Reply by rokm** [reply]
Try running `ldd` against `Mylib.so` while setting `LD_LIBRARY_PATH` to the directory where the application unpacked itself (i.e., the top-level application directory, where most .so libraries are collected). Or run `ldd` against `Mylib.so` in a sub-process from within the frozen application.
